src/components/settings-row/settings-row.test.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import { HeadingType } from "../heading";
+import SettingsRow from ".";
+
+test("renders", () => {
+  render(<SettingsRow />);
+
+  const settingsRow = screen.getByTestId("settings-row");
+  expect(settingsRow).toBeVisible();
+});
+
+test("renders with children", () => {
+  render(<SettingsRow>hello world</SettingsRow>);
+
+  const children = screen.getByText("hello world");
+  expect(children).toBeVisible();
+});
+
+test("renders with the correct default h3 heading element and text when the `title` prop is passed", () => {
+  render(<SettingsRow title="foo" />);
+
+  const heading = screen.getByRole("heading", { level: 3 });
+  expect(heading).toBeVisible();
+  expect(heading).toHaveTextContent("foo");
+});
+
+test("when the title prop is not passed, the h3 heading element does not render", () => {
+  render(<SettingsRow />);
+
+  const heading = screen.queryByRole("heading", { level: 3 });
+  expect(heading).not.toBeInTheDocument();
+});
+
+test.each([
+  ["h1", 1],
+  ["h2", 2],
+  ["h3", 3],
+  ["h4", 4],
+  ["h5", 5],
+] as [HeadingType, number][])(
+  "renders the correct heading element when the `headingType` prop is passed as %s",
+  (headingType, level) => {
+    render(<SettingsRow title="foo" headingType={headingType} />);
+
+    const heading = screen.getByRole("heading", { level });
+    expect(heading).toBeVisible();
+    expect(heading).toHaveTextContent("foo");
+  }
+);
+
+test("renders with the correct description text when the `description` prop is passed", () => {
+  render(<SettingsRow title="foo" description="bar" />);
+
+  const description = screen.getByText("bar");
+  expect(description).toBeVisible();
+});
+
+test("renders a separator element when the `description` prop is passed", () => {
+  render(<SettingsRow title="foo" description="bar" />);
+
+  const description = screen.getByText("bar");
+  expect(description).toHaveStyle({ gridRow: "3", marginTop: "0px" });
+});
+
+test("when the `description` prop is not passed, the separator does not render", () => {
```

Comment:
suggestion/question: Same thing here too. This test assertion doesn't seem to match the description. It states doesn't render a separator element but the assertion is querying if the `subtitle` is in the DOM. Is this correct?

Reply:
Yeah, good point, this is the styling we apply conditionally internally on `Heading` when the internal prop `hasSeperator` is true.

What I might do is add a `data-role` locator to the `StyledSeperator` component instead and assert against it not being in the DOM 👍